renderdoc/driver/gl/wrappers/gl_query_funcs.cpp

```diff
@@ -262,8 +262,12 @@ bool WrappedOpenGL::Serialise_glBeginQuery(GLenum target, GLuint qid)
 
   if(m_State < WRITING)
   {
-    m_Real.glBeginQuery(Target, GetResourceManager()->GetLiveResource(id).name);
-    m_ActiveQueries[QueryIdx(Target)][0] = true;
+    // Queries in the log interfere with the queries from FetchCounters.
+    if(!m_FetchCounters)
+    {
+      m_Real.glBeginQuery(Target, GetResourceManager()->GetLiveResource(id).name);
+      m_ActiveQueries[QueryIdx(Target)][0] = true;
```

Comment:
An alternative would be to not support 'Samples Written' on OpenGL as that counter is based on the occlusion query.  The pipeline statistics are less likely to be used for conditional rendering.